### ISSUE TITLE
Cython generated c files should not be part of the source distribution

### ISSUE THREAD

**Issue by ghisvail** [pyFFTW/pyFFTW#31]
From  a development standpoint, any auto-generated files should not be generated as they can be refreshed at build time. Cython should be made a build dependency in the `setup.py` file and c-files should be excluded from `MANIFEST.in`.

From a packaging standpoint, the build system will be called twice, one for the  Python 2 build and another one for Python 3. Each time, the c-files will be refreshed, overwriting the ones currently provided by the source distribution. This generates unnecessary pain for packaging.

**Comment by grlee77** [comment]
Closing this as "won't fix".  The decision was made to keep the Cython-generated .c files in the source distribution.  see https://github.com/pyFFTW/pyFFTW/pull/182#issuecomment-316913043

**Comment by jakirkham** [comment]
Interestingly we wound up adding Cython as a build requirement anyways. ( https://github.com/pyFFTW/pyFFTW/pull/177 )

**Comment by hgomersall** [comment]
I think we should reopen this in light of the changes to allowing different library versions. Further discussion on the issue should be here, but the position IMO is that we should remove the C files.

**Comment by grlee77** [comment]
I agree with removing them as well.  The recent changes for build-time detection of the available precisions makes it necessary to generate them via Cython.
